File: src/main/java/com/vaadin/demo/component/login/LoginModalOverlayHeaderCustomisation.java
package com.vaadin.demo.component.login;

import com.vaadin.flow.component.Text;
import com.vaadin.flow.component.login.LoginOverlay;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-full-source-line

@Route("login-modal-overlay-header-customisation")
public class LoginModalOverlayHeaderCustomisation extends Div {

    public LoginModalOverlayHeaderCustomisation() {
        // tag::snippet[]

        LoginOverlay loginOverlay = new LoginOverlay();
        add(loginOverlay);
        // end::snippet[]
    }

    public static class Exporter extends DemoExporter<LoginModalOverlayHeaderCustomisation> { // hidden-full-source-line
    } // hidden-full-source-line
}

File: frontend/demo/component/login/login-modal-overlay-header-customisation.ts
import '../../init'; // hidden-full-source-line

import { html, LitElement, customElement } from 'lit-element';
import '@vaadin/vaadin-login/vaadin-login-overlay';
import { applyTheme } from 'generated/theme';
import * as img from '../../../../src/main/resources/images/logo-white.png';

@customElement('login-modal-overlay-header-customisation')
export class Example extends LitElement {
  constructor() {
    super();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(this.shadowRoot);
  }

  render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-login-overlay theme="header-customised" opened description="">
        <img slot="title" src=${img} width="100" />
        <h2 slot="title" class="title">Welcome back</h2>
        <span slot="title">Don't have an account? <a href="#">Sign up here!</a></span>
      </vaadin-login-overlay>
      <!-- end::snippet[] -->
    `;
  }
}
